### PR TITLE
python310Packages.bayesian-optimization: Fix broken tests

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

Fix broken tests

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by juliendehos — APPROVED** [review]

File: pkgs/development/python-modules/bayesian-optimization/default.nix [hunk 1]
```diff
@@ -25,15 +26,29 @@ buildPythonPackage rec {
...
+    # https://github.com/fmfn/BayesianOptimization/issues/243
+    "test_suggest_with_one_observation"
+  ];
 
   meta = with lib; {
     broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
-    description = "A Python implementation of global optimization with gaussian processes";
+    description = ''
+      A Python implementation of global optimization with gaussian processes
+    '';
```
Comment (superherointj) [inline pkgs/development/python-modules/bayesian-optimization/default.nix:51]:
```suggestion
    description = "A python library implementing global optimization with gaussian processes";
```

**Comment by superherointj** [comment]
Can you add `pythonimportsCheck` too?

**Comment by superherointj** [comment]
> Is there anything that prevents this to be merged?

@onny ping

> However, I don't think it is necessary, because the test suite (fixed by this PR) imports the package as well and does much more than the import test.

It's a different test level. Fairly simple to add it.

File: pkgs/development/python-modules/bayesian-optimization/default.nix [hunk 2]
```diff
@@ -25,15 +26,29 @@ buildPythonPackage rec {
...
+    })
+  ];
+
+  checkInputs = [ pytestCheckHook ];
+
+  disabledTests = [
+    # New sklearn broke one test
+    # https://github.com/fmfn/BayesianOptimization/issues/243
+    "test_suggest_with_one_observation"
+  ];
```
Comment (superherointj) [inline pkgs/development/python-modules/bayesian-optimization/default.nix:45]:
```suggestion
  ];
  
  pythonImportsCheck = [ "bayes_opt" ];
```

Comment (superherointj) [reply pkgs/development/python-modules/bayesian-optimization/default.nix:45]: python310Packages.bayesian-optimization: add pythonImportsCheck

File: pkgs/development/python-modules/bayesian-optimization/default.nix [hunk 3]
```diff
@@ -25,15 +26,31 @@ buildPythonPackage rec {
...
+  
+  pythonImportsCheck = [ "bayes_opt" ];
```
Comment (superherointj) [inline pkgs/development/python-modules/bayesian-optimization/default.nix:47]:
```suggestion

  pythonImportsCheck = [ "bayes_opt" ];
```

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 181205` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.bayesian-optimization</li>
    <li>python39Packages.bayesian-optimization</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 181205` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.bayesian-optimization</li>
    <li>python39Packages.bayesian-optimization</li>
  </ul>
</details>

**Comment by superherointj** [comment]
When CI is done. Can squash & merge this.

**Pull request merged**